src/Core/Services/ExecutionHelper.cs

```diff
@@ -319,8 +321,8 @@ private static void SetContextResult(IMiddlewareContext context, JsonDocument? r
                 SupportedHotChocolateTypes.INT_TYPE => ((IntValueNode)value).ToInt32(),
                 SupportedHotChocolateTypes.LONG_TYPE => ((IntValueNode)value).ToInt64(),
                 SupportedHotChocolateTypes.SINGLE_TYPE => ((FloatValueNode)value).ToSingle(),
-                SupportedHotChocolateTypes.FLOAT_TYPE => ((FloatValueNode)value).ToDouble(),
-                SupportedHotChocolateTypes.DECIMAL_TYPE => ((FloatValueNode)value).ToDecimal(),
+                SupportedHotChocolateTypes.FLOAT_TYPE => value is IntValueNode intValueNode ? intValueNode.ToDouble() : ((FloatValueNode)value).ToDouble(),
+                SupportedHotChocolateTypes.DECIMAL_TYPE => value is IntValueNode intValueNode ? intValueNode.ToDecimal() : ((FloatValueNode)value).ToDecimal(),
```

Comment:
This will fix the issue if value is considered IntValueNode, but what if it is another valid float number yet, of a different value type? Would `((FloatValueNode)value)` cast fail again?

Reply:
Thanks, Ani, I tried the suggested change but the .Net float type has different range than Hotchocolate floatNodeType, which is breaking existing tests.
The float type in .NetC# has a maximum value of approximately 3.4E38. Any value larger than this will exceed the range of float, and when we try to parse such a value, C# will return Infinity, and we have a test for 2E150 which started failing.
float.Parse() for any larger value gives `Infinity`
![image](https://github.com/Azure/data-api-builder/assets/102276754/d6decab3-7757-48c1-89b4-fdfbf928f55c)

Reply:
Also noticed other tests failing like below:
![image](https://github.com/Azure/data-api-builder/assets/102276754/803be145-b2dd-4456-9417-9fb4a5de72b3)

Reply:
what are the other possibilities? i tried providing string and it gave error in hotchocolate query itself.

Reply:
how about boolean? long? short? decimal which can be parsed as float?

Reply:
I have been playing around with this today, and one option that might handle all of these cases gracefully is to have helper parsing functions for each number type that we support that needs careful conversion.

so something like

return argumentSchema.Type.TypeName().Value switch
{
    SupportedHotChocolateTypes.BYTE_TYPE => ((IntValueNode)value).ToByte(),
    SupportedHotChocolateTypes.SHORT_TYPE => ((IntValueNode)value).ToInt16(),
    SupportedHotChocolateTypes.INT_TYPE => ((IntValueNode)value).ToInt32(),
    SupportedHotChocolateTypes.LONG_TYPE => ((IntValueNode)value).ToInt64(),
    SupportedHotChocolateTypes.SINGLE_TYPE => TryParseToSingle(value),
    SupportedHotChocolateTypes.FLOAT_TYPE => TryParseToDouble(value),
    SupportedHotChocolateTypes.DECIMAL_TYPE => TryParseToDecimal(value),
    SupportedHotChocolateTypes.UUID_TYPE => Guid.TryParse(value.Value!.ToString(), out Guid guidValue) ? guidValue : value.Value,
    _ => value.Value
};

and then the `TryParseTo` helper functions can take in an `IValueNode` and based on its type, do the relevant conversion to pass back to the caller and be cast to the type we want.

Reply:
HotChocolate validation occurs before this code that ensures that the user-provided input object field value value is coercible to the type defined on the GraphQL schema. 

Using the `int_types` (integer) field as an example if the user provides:
- `"this is a string"` or `"123"` (string)
- `123.45` (decimal/float)
- `2147483648` (too large a number -> Int.MAX+1)
- `true` (boolean)

-> both BCP and the GraphQL engine will respond with
  >  "message": "The specified value type of field `int_types` does not match the field type."

Reply:
Looks like HC has only two nodes for "numbers":
- `IntValueNode` https://github.com/ChilliCream/graphql-platform/blob/12.22.4/src/HotChocolate/Language/src/Language.SyntaxTree/IntValueNode.cs
- `FloatValueNode` https://github.com/ChilliCream/graphql-platform/blob/12.22.4/src/HotChocolate/Language/src/Language.SyntaxTree/FloatValueNode.cs

This PR should be scoped to coercing userinput to the type defined for the field on the GraphQL schema. e.g when HotChocolate resolves userinput to `IntValueNode`: try to coerce that value to the appropriate `float` or `decimal` type.

I don't think we should be trying to convert a string "123.23" to a float/decimal in this PR.

Reply:
we shouldn't convert string to float/decimal, hotchocolate has its own validation based on schema generated which provide suggestions and give error when creating queries itself if we try to use different type then the target type.
Also
decimal and float are same as float in graphQL Node type
byte/short/int/long all are IntNodeType
giving boolean value will throw error for float/decimal.